controllers/machine_controller_phases.go

```diff
@@ -87,6 +87,12 @@ func (r *MachineReconciler) reconcileExternal(ctx context.Context, m *clusterv1.
 		return nil, err
 	}
 
+	// if external ref is paused, return error.
+	if util.IsPaused(cluster, obj) {
+		logger.V(3).Info("Infrastructure ref is paused")
```

Comment:
This message should be updated, this could be any type of external reference and not necessarily to an infrastructure resource.